server/api/availability.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package api
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/robfig/cron/v3"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/job"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/remote"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils/bot"
+)
+
+const (
+	AVAILABILITY_VIEW_FREE              = '0'
+	AVAILABILITY_VIEW_TENTATIVE         = '1'
+	AVAILABILITY_VIEW_BUSY              = '2'
+	AVAILABILITY_VIEW_OUT_OF_OFFICE     = '3'
+	AVAILABILITY_VIEW_WORKING_ELSEWHERE = '4'
+)
+
+type availabilityJob struct {
+	api API
+}
+
+func NewAvailabilityJob(api API) job.RecurringJob {
+	return &availabilityJob{api: api}
+}
+
+func (j *availabilityJob) Run() {
+	c := cron.New()
+	c.AddFunc("* * * * *", j.Work)
+	c.Start()
+}
+
+func (j *availabilityJob) getLogger() bot.Logger {
+	return j.api.(*api).Logger
```

Comment:
Doesn't API implement `bot.Logger`? If not, 2/5 it should, see https://github.com/levb/mattermost-plugin-solar-lottery/blob/a37a0ba0512bad58ced4a542740ec7cf9b53aa3c/server/api/api.go#L20 for how I did it there, some improvements to be back-ported.